flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/OpenApiServiceGenerator.java

```diff
@@ -233,17 +243,26 @@ public static void setGeneratedFileName(List<File> listFiles, GenSrcFile gFile,
                 gFile.getFileName().split("\\.")[1]);
     }
 
-    private void genServiceDeclaration(Path serviceObjPath, Path serviceImplPath) throws IOException,
+    private String genServiceDeclaration(GeneratedFiles generatedFileDetails,
+                                         Path serviceImplPath) throws IOException,
             WorkspaceDocumentException, EventSyncException, BallerinaOpenApiException {
-        this.workspaceManager.loadProject(projectPath.resolve(serviceObjPath));
-        Optional<SemanticModel> semanticModel =
-                this.workspaceManager.semanticModel(projectPath.resolve(serviceObjPath));
-        Optional<Document> document = this.workspaceManager.document(projectPath.resolve(serviceObjPath));
-        if (semanticModel.isEmpty() || document.isEmpty()) {
-            throw new BallerinaOpenApiException("Invalid service object is created");
-        }
-
-        TypeDefinitionSymbol symbol = getServiceTypeSymbol(semanticModel.get().moduleSymbols(), "OASServiceType");
+        Path serviceObjPath = projectPath.resolve(SERVICE_OBJ_FILE);
+        Project project = this.workspaceManager.loadProject(serviceObjPath);
+        Package currentPackage = project.currentPackage();
+        Module module = currentPackage.module(ModuleName.from(currentPackage.packageName()));
+        ModuleId moduleId = module.moduleId();
+        DocumentId serviceObjDocId = DocumentId.create(serviceObjPath.toString(), moduleId);
+        DocumentConfig documentConfig = DocumentConfig.from(
+                serviceObjDocId, generatedFileDetails.serviceObjContent(), generatedFileDetails.serviceObjFile());
+        module = module.modify().addDocument(documentConfig).apply();
+
+        DocumentId typesDocId = DocumentId.create(generatedFileDetails.typesFile(), moduleId);
+        DocumentConfig typeDocConfig = DocumentConfig.from(typesDocId, generatedFileDetails.typesContent(),
+                generatedFileDetails.typesFile());
+        module.modify().addDocument(typeDocConfig).apply();
+
+        SemanticModel semanticModel = project.currentPackage().getCompilation().getSemanticModel(moduleId);
```

Comment:
```suggestion
        SemanticModel semanticModel = currentPackage.getCompilation().getSemanticModel(moduleId);
```

Reply:
Since we change the module, we have to call `currentPackage()` again.